src/remote.c

```diff
@@ -951,15 +961,16 @@ int git_remote_fetch(
 	const git_remote_callbacks *cbs = NULL;
 	const git_strarray *custom_headers = NULL;
 	const git_proxy_options *proxy = NULL;
+	struct git_remote_connection conn;
```

Comment:
I cannot see this value ever being used. You probably meant to remove `custom_headers` and `proxy` and then call `git_remote__connect` with `conn`, didn't you?

Edit: and you also need to initialize that field, as you only conditionally assign its fields